server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1633,6 +1640,223 @@ public PrimaryDataStoreInfo cancelPrimaryStorageForMaintenance(CancelPrimaryStor
+        dataStoreVO.setPath(childStoragePoolInfo.getHostPath());
+        dataStoreVO.setPort(datastoreClusterPool.getPort());
+        dataStoreVO.setName(childStoragePoolInfo.getName());
+        dataStoreVO.setUuid(childStoragePoolInfo.getUuid());
+        dataStoreVO.setDataCenterId(datastoreClusterPool.getDataCenterId());
+        dataStoreVO.setPodId(datastoreClusterPool.getPodId());
+        dataStoreVO.setClusterId(datastoreClusterPool.getClusterId());
+        dataStoreVO.setStatus(StoragePoolStatus.Up);
+        dataStoreVO.setUserInfo(datastoreClusterPool.getUserInfo());
+        dataStoreVO.setManaged(datastoreClusterPool.isManaged());
+        dataStoreVO.setCapacityIops(datastoreClusterPool.getCapacityIops());
+        dataStoreVO.setCapacityBytes(childDataStoreAnswer.getPoolInfo().getCapacityBytes());
+        dataStoreVO.setUsedBytes(childDataStoreAnswer.getPoolInfo().getCapacityBytes() - childDataStoreAnswer.getPoolInfo().getAvailableBytes());
+        dataStoreVO.setHypervisor(datastoreClusterPool.getHypervisor());
+        dataStoreVO.setScope(datastoreClusterPool.getScope());
+        dataStoreVO.setParent(datastoreClusterPool.getId());
+
+        Map<String, String> details = new HashMap<>();
+        if(org.apache.commons.lang.StringUtils.isNotEmpty(childDataStoreAnswer.getPoolType())) {
+            details.put("pool_type", childDataStoreAnswer.getPoolType());
+        }
+        _storagePoolDao.persist(dataStoreVO, details, storageTags);
+        return dataStoreVO;
+    }
+
+    private void handleRemoveChildStoragePoolFromDatastoreCluster(Set<String> childDatastoreUUIDs) {
+
+        for (String childDatastoreUUID : childDatastoreUUIDs) {
+            StoragePoolVO dataStoreVO = _storagePoolDao.findPoolByUUID(childDatastoreUUID);
+            List<VolumeVO> allVolumes = _volumeDao.findByPoolId(dataStoreVO.getId());
+
+            for (VolumeVO volume : allVolumes) {
+                VMInstanceVO vmInstance = _vmInstanceDao.findById(volume.getInstanceId());
+                if (vmInstance == null) {
+                    continue;
+                }
+                long volumeId = volume.getId();
+                Long hostId = vmInstance.getHostId();
+                if (hostId == null) {
+                    hostId = vmInstance.getLastHostId();
+                }
+
+                // Prepare for the syncvolumepath command
+                DataTO volTO = volFactory.getVolume(volume.getId()).getTO();
+                DiskTO disk = new DiskTO(volTO, volume.getDeviceId(), volume.getPath(), volume.getVolumeType());
+                Map<String, String> details = new HashMap<String, String>();
+                details.put(DiskTO.PROTOCOL_TYPE, Storage.StoragePoolType.DatastoreCluster.toString());
+                disk.setDetails(details);
+
+                SyncVolumePathCommand cmd = new SyncVolumePathCommand(disk, vmInstance.getInstanceName());
+                final Answer answer = _agentMgr.easySend(hostId, cmd);
```

Comment:
Is this allowed for disabled host ?

Reply:
Yes, it works. I've tested with a disabled host. The command finally goes to vCenter.

Reply:
@harikrishna-patnala OK, can you add a log to indicate the volume syncing for the disable host(s)?

Reply:
Added log for the same.